File: Problems 3100-3199/3136. Valid Word/solution.ts
const VOWELS = new Set(['a', 'e', 'o', 'i', 'u']);
function isValid(word: string): boolean {
    // if (word.length < 3) return false;
    // return /^[a-zA-Z0-9]{3,}$/.test(word);
    if (!/^[a-z0-9]{3,}$/i.test(word)) return false;
    for (const c of word) {
        if (VOWELS.has(c.toLowerCase())) {
            return /[b-df-hj-np-tv-z]+/i.test(word);
        }
    }
    return false;
};

File: Problems 3100-3199/3136. Valid Word/solution.py
class Solution:
    def isValid(self, word: str) -> bool:
        if len(word) < 3:
            return False
        
        vowels = 0
        consonants = 0

        for c in word:
            if c.isalpha():
                if c.lower() in 'aeiou':
                    vowels += 1
                else:
                    consonants += 1
            elif not c.isdigit():
                return False
        
        if vowels and consonants:
            return True
        return False